tools/crm_rule.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (C) 2019 Chris Lumens <clumens@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <crm_internal.h>
+
+#include <crm/cib.h>
+#include <crm/common/iso8601.h>
+#include <crm/common/xml.h>
+#include <crm/crm.h>
+#include <crm/msg_xml.h>
+#include <crm/pengine/rules.h>
+#include <crm/pengine/rules_internal.h>
+#include <crm/pengine/status.h>
+
+#include <sys/stat.h>
+
+#define CMD_ERR(fmt, args...) do {          \
+        crm_warn(fmt, ##args);              \
+        fprintf(stderr, fmt "\n", ##args);  \
+    } while(0)
+
+enum crm_rule_mode {
+    crm_rule_mode_none,
+    crm_rule_mode_check
+} rule_mode = crm_rule_mode_none;
+
+static int crm_rule_check(cib_t *cib_conn, xmlNode *inputXml, const char *rule_id, crm_time_t *effective_date);
+
+static struct crm_option long_options[] = {
+    /* Top-level Options */
+    {"help",       no_argument,       NULL, '?', "\tThis text"},
+    {"version",    no_argument,       NULL, '$', "\tVersion information"  },
+    {"verbose",    no_argument,       NULL, 'V', "\tIncrease debug output"},
+
+    {"-spacer-",   required_argument, NULL, '-', "\nModes (mutually exclusive):" },
+    {"check",      no_argument,       NULL, 'c', "\tCheck whether a rule is in effect" },
+
+    {"-spacer-",   required_argument, NULL, '-', "\nAdditional options:" },
+    {"date",       required_argument, NULL, 'd', "Whether the rule is in effect on a given date" },
+    {"rule",       required_argument, NULL, 'r', "The ID of the rule to check" },
+
+    {"-spacer-",   no_argument,       NULL, '-', "\nData:"},
+    {"xml-text",   required_argument, NULL, 'X', "Retrieve XML from the supplied string"},
+    {"xml-file",   required_argument, NULL, 'x', "Retrieve XML from the named file"},
+    {"xml-pipe",   no_argument,       NULL, 'p', "Retrieve XML from stdin"},
```

Comment:
Let's add a footer (like the crm_resource "Examples" section) that states: "This tool is currently experimental. The interface, behavior, and output may change with any version of pacemaker."